akka-stream/src/main/scala/akka/stream/Graph.scala

```diff
@@ -28,9 +29,29 @@ trait Graph[+S <: Shape, +M] {
    */
   private[stream] def traversalBuilder: TraversalBuilder
 
+  @deprecated("Use addAttributes instead of withAttributes, will be made internal", "2.5.8")
   def withAttributes(attr: Attributes): Graph[S, M]
 
-  def named(name: String): Graph[S, M] = addAttributes(Attributes.name(name))
+  /**
+   * Add a name for this graph, if this node or composed node already has a name it will be replaced.
+   */
+  def named(name: String): Graph[S, M] = {
+    traversalBuilder.attributes.get[Attributes.Name] match {
+      case Some(previous) if previous.n != name ⇒
```

Comment:
what happens when `.named("foo").named("foo")` ?

Reply:
Good catch, didn't think of that one